system-test/test-e2e.ts

```diff
@@ -141,307 +136,218 @@ describe('@google-cloud/debug end-to-end behavior', () => {
-          return Promise.all([breakpoint, delay(10 * 1000)]);
-        })
-        .then((results: stackdriver.Breakpoint[]) => {
-          // Check that the contents of the log is correct
-
-          const breakpoint = results[0];
-
-          // If no throttling occurs, we expect ~20 logs since we are logging
-          // 2x per second over a 10 second period.
-          children.forEach((child) => {
-            const logCount = (child.transcript.match(
-                                  /LOGPOINT: o is: \{"a":\[1,"hi",true\]\}/g) ||
-                              []).length;
-            // A log count of greater than 10 indicates that we did not
-            // successfully pause when the rate of `maxLogsPerSecond` was
-            // reached.
-            assert(logCount <= 10, 'log count is greater than 10: ' + logCount);
-            // A log count of less than 3 indicates that we did not successfully
-            // resume logging after `logDelaySeconds` have passed.
-            assert(
-                logCount > 2, 'log count is not greater than 2: ' + logCount);
-          });
-
-          // TODO: Determine how to properly specify the signature of
-          // deleteBreakpoint
-          return (api as {deleteBreakpoint: Function})
-              .deleteBreakpoint(debuggeeId, breakpoint.id);
-        })
-        .then(() => {
-          console.log('-- test passed');
-          return Promise.resolve();
-        });
+
+    const debuggees = await api.listDebuggees(projectId!, true);
+
+    // Check that the debuggee created in this test is among the list of
+    // debuggees, then list its breakpoints
+    console.log(
+        '-- List of debuggees\n', util.inspect(debuggees, {depth: null}));
+    assert.ok(debuggees, 'should get a valid ListDebuggees response');
+
+    const result = _.find(debuggees, (d: Debuggee) => {
+      return d.id === debuggeeId;
+    });
+    assert.ok(result, 'should find the debuggee we just registered');
+
+    const breakpoints = await api.listBreakpoints(debuggeeId!, {});
+    console.log('-- List of breakpoints\n', breakpoints);
+
+    // Delete every breakpoint
+    const promises = breakpoints.map((breakpoint: stackdriver.Breakpoint) => {
+      return api.deleteBreakpoint(debuggeeId!, breakpoint.id);
+    });
+
+    await Promise.all(promises);
+
+    const foundBreakpoints = await api.listBreakpoints(debuggeeId!, {});
+    assert.strictEqual(foundBreakpoints.length, 0);
+    console.log('-- deleted');
+
+    // Set a breakpoint at which the debugger should write to a log
+    console.log('-- setting a logpoint');
+    const breakpoint = await api.setBreakpoint(debuggeeId!, {
+      id: 'breakpoint-3',
+      location: {path: FILENAME, line: 5},
+      condition: 'n === 10',
+      action: 'LOG',
+      expressions: ['o'],
+      logMessageFormat: 'o is: $0',
+      stackFrames: [],
+      evaluatedExpressions: [],
+      variableTable: []
+    });
```

Comment:
Rename `verifySetLogpoint` to `verifySetBreakpoint` and make it accept a parameter – the breakpoint. Reuse the function to replace the code here.

Reply:
Done